--- occu/WebUI/www/rega/esp/controls/heating_control.fn.orig
+++ occu/WebUI/www/rega/esp/controls/heating_control.fn
@@ -429,75 +429,75 @@
         WriteLine("</tr>");
 
 
-      ! HEATING_COOLING
-      if (oHeatingCooling) {
-        string btnStateHeating = "ControlBtnOn";
-        string btnStateCooling = "ControlBtnOff";
-
-        string statusHeating = "${stringTableHeating}";
-        string statusCooling = "${stringTableCooling}";
-
-        if (oHeatingCooling.Value() == 1) {
-          btnStateHeating = "ControlBtnOff";
-          btnStateCooling = "ControlBtnOn";
+        ! HEATING_COOLING
+        if (oHeatingCooling) {
+          string btnStateHeating = "ControlBtnOn";
+          string btnStateCooling = "ControlBtnOff";
+
+          string statusHeating = "${stringTableHeating}";
+          string statusCooling = "${stringTableCooling}";
+
+          if (oHeatingCooling.Value() == 1) {
+            btnStateHeating = "ControlBtnOff";
+            btnStateCooling = "ControlBtnOn";
+          }
+
+          WriteLine("<tr>");
+            WriteLine("<td style='width:20%;'>");
+              WriteLine("<div class='"#btnStateHeating#" CLASS02551' style='line-height:35px;' id='"#chnId#"Heating' onclick='setDpState("#oHeatingCooling.ID()#", 0); JControlBtn.pressed(jQuery(this));'>"#statusHeating#"</div>");
+            WriteLine("</td>");
+
+            WriteLine("<td><div><img src='/ise/img/help.png' style='cursor: pointer; width:25px; height:25px;' onclick='showParamHelp(translateKey(\"helpHeatingCooling\"), 450, 150);'></div></td>");
+
+            WriteLine("<td>");
+              WriteLine("<div class='"#btnStateCooling#" CLASS02551' style='line-height:35px;' id='"#chnId#"Cooling' onclick='setDpState("#oHeatingCooling.ID()#", 1); JControlBtn.pressed(jQuery(this));'>"#statusCooling#"</div>");
+            WriteLine("</td>");
+          WriteLine("</tr>");
+         ! END HEATING_COOLING
         }
-        ! END HEATING_COOLING
-
         WriteLine("<tr>");
-          WriteLine("<td style='width:20%;'>");
-            WriteLine("<div class='"#btnStateHeating#" CLASS02551' style='line-height:35px;' id='"#chnId#"Heating' onclick='setDpState("#oHeatingCooling.ID()#", 0); JControlBtn.pressed(jQuery(this));'>"#statusHeating#"</div>");
-          WriteLine("</td>");
-
-          WriteLine("<td><div><img src='/ise/img/help.png' style='cursor: pointer; width:25px; height:25px;' onclick='showParamHelp(translateKey(\"helpHeatingCooling\"), 450, 150);'></div></td>");
-
-          WriteLine("<td>");
-            WriteLine("<div class='"#btnStateCooling#" CLASS02551' style='line-height:35px;' id='"#chnId#"Cooling' onclick='setDpState("#oHeatingCooling.ID()#", 1); JControlBtn.pressed(jQuery(this));'>"#statusCooling#"</div>");
+          WriteLine("<td colspan='3'>");
+            WriteLine("<div class='ControlBtnOff'>");
+              WriteLine("<span>${weeklyProfileLbl}</span>");
+              WriteLine("<select id='"#chnId#"ActiveProfile' style='margin-left:15%;' onchange='setDpState("#oActiveProfile.ID()#",this.value)'>");
+                WriteLine("<option value='1'>${optionWeeklyProfile1}</option>");
+                WriteLine("<option value='2'>${optionWeeklyProfile2}</option>");
+                WriteLine("<option value='3'>${optionWeeklyProfile3}</option>");
+
+                ! Some devices (eTRV and WTH with a firmware < 1.6 and others) do have only 3 weekly programs while others have 6 of them.
+                if (
+                  (
+                    (chnLabel != "HMIP-eTRV")
+                    && (chnLabel != "HmIP-eTRV")
+                    && (chnLabel != "HMIP-eTRV-2")
+                    && (chnLabel != "HmIP-eTRV-2")
+                    && (chnLabel != "HmIP-eTRV-2 I9F")
+                    && (chnLabel != "HmIP-eTRV-2-UK")
+                    && (chnLabel != "HmIP-eTRV-B")
+                    && (chnLabel != "HmIP-eTRV-B-2")
+                    && (chnLabel != "HmIP-eTRV-B-2 R4M")
+                    && (chnLabel != "HmIP-eTRV-B-UK")
+                    && (chnLabel != "HmIP-eTRV-B1")
+                    && (chnLabel != "HmIP-WTH")
+                    && (chnLabel != "HmIP-WTH-A")
+                    && (chnLabel != "HMIP-WTH")
+                    && (chnLabel != "HmIP-eTRV-C")
+                    && (chnLabel != "HmIP-eTRV-C-2")
+                    && (chnLabel != "HmIP-eTRV-CL")
+                    && (chnLabel != "HmIP-eTRV-E")
+                    && (chnLabel != "HmIP-eTRV-E-S")
+                    && (chnLabel != "HmIP-eTRV-E-A")
+                  )
+                  || (((chnLabel == "HmIP-WTH") || (chnLabel == "HMIP-WTH") || (chnLabel == "HmIP-WTH-A")) && ((devFwMajor > 1) || ((devFwMajor == 1) && (devFwMinor >= 6))))
+                 ) {
+                    WriteLine("<option value='4'>${optionWeeklyProfile4}</option>");
+                    WriteLine("<option value='5'>${optionWeeklyProfile5}</option>");
+                    WriteLine("<option value='6'>${optionWeeklyProfile6}</option>");
+                }
+              WriteLine("</select>");
+            WriteLine("</div>");
           WriteLine("</td>");
-            WriteLine("<tr>");
-              WriteLine("<td colspan='3'>");
-                WriteLine("<div class='ControlBtnOff'>");
-                  WriteLine("<span>${weeklyProfileLbl}</span>");
-                  WriteLine("<select id='"#chnId#"ActiveProfile' style='margin-left:15%;' onchange='setDpState("#oActiveProfile.ID()#",this.value)'>");
-                    WriteLine("<option value='1'>${optionWeeklyProfile1}</option>");
-                    WriteLine("<option value='2'>${optionWeeklyProfile2}</option>");
-                    WriteLine("<option value='3'>${optionWeeklyProfile3}</option>");
-
-                    ! Some devices (eTRV and WTH with a firmware < 1.6 and others) do have only 3 weekly programs while others have 6 of them.
-                    if (
-                      (
-                        (chnLabel != "HMIP-eTRV")
-                        && (chnLabel != "HmIP-eTRV")
-                        && (chnLabel != "HMIP-eTRV-2")
-                        && (chnLabel != "HmIP-eTRV-2")
-                        && (chnLabel != "HmIP-eTRV-2 I9F")
-                        && (chnLabel != "HmIP-eTRV-2-UK")
-                        && (chnLabel != "HmIP-eTRV-B")
-                        && (chnLabel != "HmIP-eTRV-B-2")
-                        && (chnLabel != "HmIP-eTRV-B-2 R4M")
-                        && (chnLabel != "HmIP-eTRV-B-UK")
-                        && (chnLabel != "HmIP-eTRV-B1")
-                        && (chnLabel != "HmIP-WTH")
-                        && (chnLabel != "HmIP-WTH-A")
-                        && (chnLabel != "HMIP-WTH")
-                        && (chnLabel != "HmIP-eTRV-C")
-                        && (chnLabel != "HmIP-eTRV-C-2")
-                        && (chnLabel != "HmIP-eTRV-CL")
-                        && (chnLabel != "HmIP-eTRV-E")
-                        && (chnLabel != "HmIP-eTRV-E-S")
-                        && (chnLabel != "HmIP-eTRV-E-A")
-                      )
-                      || (((chnLabel == "HmIP-WTH") || (chnLabel == "HMIP-WTH") || (chnLabel == "HmIP-WTH-A")) && ((devFwMajor > 1) || ((devFwMajor == 1) && (devFwMinor >= 6))))
-                     ) {
-                        WriteLine("<option value='4'>${optionWeeklyProfile4}</option>");
-                        WriteLine("<option value='5'>${optionWeeklyProfile5}</option>");
-                        WriteLine("<option value='6'>${optionWeeklyProfile6}</option>");
-                    }
-                  WriteLine("</select>");
-                WriteLine("</div>");
-              WriteLine("</td>");
-            WriteLine("</tr>");
         WriteLine("</tr>");
-      }
       WriteLine("</table>");
     WriteLine("</td>");
 
--- occu/WebUI/www/rega/esp/functions.fn.orig
+++ occu/WebUI/www/rega/esp/functions.fn
@@ -816,7 +816,7 @@
     }
   }
 
-  if (oCC.HssType() == "HEATING_CLIMATECONTROL_TRANSCEIVER") {
+  if ((oCC.HssType() == "HEATING_CLIMATECONTROL_TRANSCEIVER") && (oCC.Label().Find("HmIPW-WGTC") != -1)) {
     if (oDP.HSSID().Find("ACTIVE_PROFILE") != -1) {
       showChannelParam = false;
     }
@@ -1018,7 +1018,7 @@
     }
   }
 
-  if (oCH.HssType().Find("HEATING_CLIMATECONTROL_TRANSCEIVER") != -1) {
+  if ((oCH.HssType().Find("HEATING_CLIMATECONTROL_TRANSCEIVER") != -1) && (oCC.Label().Find("HmIPW-WGTC") != -1)) {
     if (
     (oDP.HSSID().Find("ACTIVE_PROFILE") != -1)
     || (oDP.HSSID().Find("DURATION_VALUE") != -1)
